### PR TITLE
 add analysisframework roles query

### PR DESCRIPTION
**Pull request by sudan45**
Addresses:
- https://github.com/the-deep/server/issues/1473

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] n+1 queries
- [x] flake8 issues
- [x] `print`
- [x] typos
- [x] unwanted comments

## This PR contains valid:

- [x] tests
- [ ] permission checks (tests here too)
- [ ] translations

### CODE REVIEW

File: apps/analysis_framework/enums.py [hunk 1]
```diff
@@ -10,8 +10,8 @@
 )
 
 
-AnalysisFrameworkRoleTypeEnum = convert_enum_to_graphene_enum(
-    AnalysisFrameworkRole.Type, name='AnalysisFrameworkRoleTypeEnum')
+AnalysisFrameworkRolesTypeEnum = convert_enum_to_graphene_enum(
+    AnalysisFrameworkRole.Type, name='AnalysisFrameworkRolesTypeEnum')
```
Comment (thenav56) [inline apps/analysis_framework/enums.py]: Type name shouldn't be plural.
```suggestion
AnalysisFrameworkRoleTypeEnum = convert_enum_to_graphene_enum(
    AnalysisFrameworkRole.Type, name='AnalysisFrameworkRoleTypeEnum')
```

File: apps/analysis_framework/filter_set.py [hunk 2]
```diff
@@ -86,3 +87,16 @@ def filter_recently_used(self, queryset, name, value):
...
+class AnalysisFrameworkRoleGqFilterSet(UserResourceGqlFilterSet):
+    search = django_filters.CharFilter(method='search_filter')
+
+    class Meta:
+        model = AnalysisFrameworkRole
+        fields = ('__all__')
+
+    def search_filter(self, qs, _, value):
+        if value:
+            return qs.filter(models.Q(title__icontains=value))
```
Comment (thenav56) [inline apps/analysis_framework/filter_set.py]:
```suggestion
            return qs.filter(title__icontains=value)
```

File: apps/analysis_framework/schema.py [hunk 3]
```diff
@@ -158,10 +158,9 @@ def resolve_used_in_project_count(root, info):
         return info.context.dl.analysis_framework.af_project_count.load(root.id)
 
 
-class AnalysisFrameworkRoleType(DjangoObjectType):
+class AnalysisFrameworkRolesListType(DjangoObjectType):
```
Comment (thenav56) [inline apps/analysis_framework/schema.py]: Type name shouldn't be plural
```suggestion
class AnalysisFrameworkRoleListType(DjangoObjectType):
```

File: apps/analysis_framework/schema.py [hunk 4]
```diff
@@ -158,10 +158,9 @@ def resolve_used_in_project_count(root, info):
...
     class Meta:
         model = AnalysisFrameworkRole
-        only_fields = ('id', 'title',)
```
Comment (thenav56) [inline apps/analysis_framework/schema.py:164]: Why are we removing this?

File: apps/analysis_framework/schema.py [hunk 5]
```diff
@@ -311,6 +310,7 @@ class Query:
             page_size_query_param='pageSize'
         )
     )
+    analysis_framework_roles = graphene.List(graphene.NonNull(AnalysisFrameworkRolesListType))
```
Comment (thenav56) [inline apps/analysis_framework/schema.py]: In general, we use `....ListType` for pagination. For this, this should be
```suggestion
    analysis_framework_roles = graphene.List(graphene.NonNull(AnalysisFrameworkRoleType), required=True)
```

File: apps/analysis_framework/tests/test_schemas.py [hunk 6]
```diff
@@ -323,3 +325,24 @@ def test_recent_analysis_framework(self):
...
         )
+
+    def test_analysis_framework_roles(self):
+        query = '''
+            query MyQuery {
+              analysisFrameworkRoles {
+                isPrivateRole
+                isDefaultRole
+                title
+                type
```
Comment (thenav56) [inline apps/analysis_framework/tests/test_schemas.py]: Let's use all the available fields

**Review by thenav56 — CHANGES_REQUESTED** [review]

File: apps/analysis_framework/filter_set.py [hunk 7]
```diff
@@ -86,3 +87,16 @@ def filter_recently_used(self, queryset, name, value):
...
+    search = django_filters.CharFilter(method='search_filter')
+
+    class Meta:
+        model = AnalysisFrameworkRole
+        fields = ('__all__')
+
+    def search_filter(self, qs, _, value):
+        if value:
+            return qs.filter(title__icontains=value)
+        return qs
```
Comment (thenav56) [inline apps/analysis_framework/filter_set.py]: Not used or required for a bound data.

**Review by thenav56 — CHANGES_REQUESTED** [review]
Looks good. Let's clean-up not required codes.

File: apps/analysis_framework/filter_set.py (hunk 7 above)
Comment (sudan45) [reply apps/analysis_framework/filter_set.py]: `AnalysisFrameworkRoleGqFilterSet` is not used 
i have to remove it

**Review by thenav56 — APPROVED** [review]

**Pull request merged**